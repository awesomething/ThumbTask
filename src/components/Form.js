import React from "react";
import config from '../config'

const Form = ({setInputText, todos, setTodos, inputText, setStatus}) => {

const postTodos = (title, categories) =>{
  return fetch(`${config.API_ENDPOINT}/todos`, {
    method: 'POST',
    headers: {
      'content-type': 'application/json',
    },
    body: JSON.stringify({
      title,
      categories,
    }),
  })
    .then(res =>
      (!res.ok)
        ? res.json().then(e => Promise.reject(e))
        : res.json()
    )
}
//
    const inputTextHandler = (e) => {
     
     setInputText(e.target.value);
    };
    const submitTodoHandler = (e) => {
        e.preventDefault();
        postTodos(inputText)
        setTodos([
            ...todos, {text: inputText, completed: false, id: Math.random() * 1000 },
        ]);
        setInputText("");
    };
    const statusHandler = (e) => {
        setStatus(e.target.value);
    }

    return (
        <form onSubmit={submitTodoHandler}>
      <input 
      value={inputText} 
      onChange={inputTextHandler} 
      required
      type="text" 
      className="todo-input" />

      <button className="todo-button" type="submit">
        <i className="fas fa-plus-square"></i>
      </button>
      {/* <div className="select">
        <select onChange={statusHandler} name="todos" className="filter-todo">
          <option value="all">All</option>
          <option value="completed">Completed</option>
          <option value="uncompleted">Uncompleted</option>
        </select>
      </div> */}
    </form>
    );
}

export default Form;