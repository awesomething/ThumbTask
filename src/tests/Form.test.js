import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Form from '../components/Form'
import {configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })



describe(`Form component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
    
  }

  it('renders a Form by default', () => {
    const wrapper = shallow(<Form />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders the Form given props', () => {
    const wrapper = shallow(<Form {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
