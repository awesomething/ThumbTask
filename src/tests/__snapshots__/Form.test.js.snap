// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders a Form by default 1`] = `
<form>
  <input
    classNameName="todo-input"
    onChange={[Function]}
    type="text"
  />
  <button
    className="todo-button"
    onClick={[Function]}
    type="submit"
  >
    <i
      className="fas fa-plus-square"
    />
  </button>
  <div
    className="select"
  >
    <select
      className="filter-todo"
      name="todos"
      onChange={[Function]}
    >
      <option
        value="all"
      >
        All
      </option>
      <option
        value="completed"
      >
        Completed
      </option>
      <option
        value="uncompleted"
      >
        Uncompleted
      </option>
    </select>
  </div>
</form>
`;

exports[`Form component renders the Form given props 1`] = `
<form>
  <input
    classNameName="todo-input"
    onChange={[Function]}
    type="text"
  />
  <button
    className="todo-button"
    onClick={[Function]}
    type="submit"
  >
    <i
      className="fas fa-plus-square"
    />
  </button>
  <div
    className="select"
  >
    <select
      className="filter-todo"
      name="todos"
      onChange={[Function]}
    >
      <option
        value="all"
      >
        All
      </option>
      <option
        value="completed"
      >
        Completed
      </option>
      <option
        value="uncompleted"
      >
        Uncompleted
      </option>
    </select>
  </div>
</form>
`;
